server/pulp/server/webservices/views/serializers/__init__.py

```diff
@@ -0,0 +1,163 @@
+import copy
+
+from django.core.urlresolvers import reverse
+
+
+class BaseSerializer(object):
+    """
+    Base class to be used for creating serializers
+    """
+
+    def __init__(self, instance=None, context=None, multiple=False):
```

Comment:
On 06/04/2015 03:52 PM, Barnaby Court wrote:
> Unfortunately I know of no absolute way to detect iterability that
> doesn't leave the possibility of false positives. For example, a string
> is iterable even though you probably wouldn't consider it to be iterable
> from a serialization standpoint. The cleanest thing is to ask the user
> of the class to state their intent.

The collections module has an abstract base class called Iterable that
matches a list and a string:

```
>>> import collections
>>> isinstance([1, 2, 3], collections.Iterable)
True
>>> isinstance('hello', collections.Iterable)
True
>>> isinstance(5, collections.Iterable)
False
```

Or was it that you don't want a string to count as an iterable? If
that's the case, you can test this way:

```
>>> hasattr('hello', '__iter__')
False
>>> hasattr([1, 2, 3], '__iter__')
True
```

-- 
Randy Barlow